src/editor/interaction/resize-controller.ts

```diff
@@ -0,0 +1,193 @@
+    this.resizeContainer.appendChild(this.resizeBorderBottom);
+    this.resizeContainer.appendChild(this.resizeAnchorLeftTop);
+    this.resizeContainer.appendChild(this.resizeAnchorLeftBottom);
+    this.resizeContainer.appendChild(this.resizeAnchorRightTop);
+    this.resizeContainer.appendChild(this.resizeAnchorRightBottom);
+    this.container.appendChild(this.resizeContainer);
+  }
+
+  // eslint-disable-next-line
+  addResizeListener(listener: (event: any, nextRect: IRect) => void) {
+    this.resizeListeners = this.resizeListeners.concat(listener);
+  }
+
+  // eslint-disable-next-line
+  removeResizeListener(listener: (event: any, nextRect: IRect) => void) {
+    this.resizeListeners = this.resizeListeners.filter(lastListener => lastListener !== listener);
+  }
+
+  // eslint-disable-next-line
+  addResizeEndListener(listener: (event: any, nextRect: IRect) => void) {
+    this.resizeEndListeners = this.resizeEndListeners.concat(listener);
+  }
+
+  // eslint-disable-next-line
+  removeResizeEndListener(listener: (event: any, nextRect: IRect) => void) {
+    this.resizeEndListeners = this.resizeEndListeners.filter(lastListener => lastListener !== listener);
+  }
+
+  release() {
+    this.editor.getContainer().removeEventListener('pointermove', this.onDragging);
+    this.editor.getContainer().removeEventListener('pointerup', this.onDragEnd);
+    this.container.removeChild(this.resizeContainer);
+  }
+
+  private update(rect: IRect) {
+    this.resizeContainer.style.left = rect.x + 'px';
+    this.resizeContainer.style.top = rect.y + 'px';
+    this.resizeContainer.style.width = rect.width + 'px';
+    this.resizeContainer.style.height = rect.height + 'px';
+    this.rect = Object.assign({}, rect);
+  }
+
+  private onDragStart = (resizeMode: number) => (event: MouseEvent) => {
+    this.isDragging = true;
+    this.resizeMode = resizeMode;
+    this.startPoint = { x: event.clientX, y: event.clientY };
+    this.startRect = Object.assign({}, this.rect);
+    event.stopPropagation();
+  };
+
+  private onDragging = (event: MouseEvent) => {
+    if (this.isDragging && event) {
+      const currentPoint = { x: event.clientX, y: event.clientY };
+      const nextRect = this.computeRect(currentPoint);
+      this.update(nextRect);
+      this.resizeListeners.forEach(listener => listener.call(null, event, nextRect));
+    }
+    event.stopPropagation();
+  };
+
+  private onDragEnd = (event: MouseEvent) => {
+    if (this.isDragging && event) {
+      const currentPoint = { x: event.clientX, y: event.clientY };
+      const nextRect = this.computeRect(currentPoint);
+      this.update(nextRect);
+      this.resizeEndListeners.forEach(listener => listener.call(null, event, nextRect));
+      this.isDragging = false;
+    }
+    event.stopPropagation();
+  };
+
+  private computeRect(point: IPointLike) {
+    const nextRect = Object.assign({}, this.startRect);
+    // left
+    if (this.resizeMode & 0b10000) {
```

Comment:
That's not very clear, how about a constant instead?
like:
```
const LEFT = '0b10000'
if (this.resizeMode & LEFT)
```

Reply:
Rejected. Next one.